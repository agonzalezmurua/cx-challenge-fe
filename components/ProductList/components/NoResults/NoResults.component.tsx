import { selectQuery } from "@/redux/app.slice";
import { useAppSelector } from "@/redux/store";
import { useTranslation } from "next-i18next";
import styles from "./NoResults.module.scss";

export const NoResults = () => {
  const query = useAppSelector(selectQuery);
  const { t } = useTranslation();

  return (
    <section className={styles.container}>
      <section className={styles.no_results__ilustration}>🌧️</section>
      <h1 className={styles.no_results__title}>
        {t("product_list.empty_result_with_query.title")}
      </h1>
      <p className={styles.no_results__label}>
        {t("product_list.empty_result_with_query.label")}
      </p>
    </section>
  );
};
