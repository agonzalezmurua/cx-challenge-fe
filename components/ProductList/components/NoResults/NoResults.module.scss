.container {
  flex: 1;
  display:flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  gap: 2rem;
}

.no_results__ilustration {
  font-size: 15rem;
}

.no_results__title {
  font-size: xx-large;
}

.no_results__label {
  font-size: medium;
}