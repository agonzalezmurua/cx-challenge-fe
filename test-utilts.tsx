import { makeStore } from "@/redux/store";
import { queries, Queries } from "@testing-library/dom";
import { render, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";

const store = makeStore({
  app: {
    parameters: {
      sorts: [
        {
          id: "irrelevant",
          name: "Menos relevante",
        },
      ],
      filters: [
        {
          id: "most_expensive",
          name: "Más caro",
          type: "price",
          values: [
            {
              id: "*-*",
              name: "infinito a infinito (infinito)",
              results: 10,
            },
          ],
        },
      ],
    },
    products: [
      {
        address: {
          city_name: "Caballito",
          state_name: "Capital Federal",
        },
        condition: "new",
        free_shipping: true,
        title: "Apple AirPods Con Estuche De Carga - Blanco",
        id: "MLA1363069275",
        installments: {
          amount: "6674.27",
          quantity: 6,
        },
        picture:
          "http://http2.mlstatic.com/D_923638-MLA54361048207_032023-I.jpg",
        price: {
          currency: "ARS",
          amount: "26994",
          decimals: 0,
        },
      },
    ],
    query: {
      price: "*-10000",
      search: "Apple Ipod",
      sort: { id: "relevance", name: "Más relevante" },
    },
  },
});

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = <
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
