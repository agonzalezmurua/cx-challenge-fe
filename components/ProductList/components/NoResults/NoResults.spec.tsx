import { NoResults } from "./NoResults.component";
import { render } from "@/test-utilts";

it("should be defined", () => {
  expect(NoResults).toBeDefined();
});

it("should render and match snapshot", () => {
  const result = render(<NoResults />);

  expect(result.asFragment()).toMatchSnapshot();
});
