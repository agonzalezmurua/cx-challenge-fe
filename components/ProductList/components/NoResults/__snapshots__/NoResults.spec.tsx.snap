// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <section
      class="no_results__ilustration"
    >
      🌧️
    </section>
    <h1
      class="no_results__title"
    >
      product_list.empty_result_with_query.title
    </h1>
    <p
      class="no_results__label"
    >
      product_list.empty_result_with_query.label
    </p>
  </section>
</DocumentFragment>
`;
